src/app.jsx

```diff
@@ -12,6 +13,55 @@ import routes from './routes';
 import analytics from './services/analytics';
 
 const App = () => {
+  // Code for setting up color schemes
+
+  const { mode, setMode } = useColorScheme();
+  // mode listener code from
+  //https://dev.to/gustavogomez/manage-dark-mode-in-reactjs-with-listener-to-detect-dark-mode-in-the-browser-3120
+
+  let colorMode = localStorage.getItem('colorMode') ?? 'system';
+
+  const onSelectMode = (prefersDark) => {
+    prefersDark === 'dark' ? setMode('dark') : setMode('light');
+  };
+
+  const setupColorSchemeListeners = (colorScheme) => {
+    if (colorMode === 'system' || colorMode === null) {
+      //anything other than 'dark' will set the scheme to light
+      setMode(colorMode);
+
+      //listen for system setting changes
+      window
+        .matchMedia('(prefers-color-scheme: dark)')
+        .addEventListener('change', (e) => onSelectMode(e.matches ? 'dark' : 'light'));
+    } else {
+      //stop listening for system setting changes and set the mode
+      window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', () => {});
+      setMode(colorMode);
+    }
+  };
+
+  //Callback, when the colorScheme changes in local storage
+  const onModeUpdate = () => {
+    setupColorSchemeListeners(localStorage.getItem('colorMode'));
+  };
+
+  useEffect(() => {
+    //Listen for changes in local storage
+    window.addEventListener('storage', onModeUpdate);
+    setupColorSchemeListeners(colorMode);
+    return () => {
+      //Clean up listeners
+      window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', () => {});
```

Comment:
This will not work, you have to pass the same function to `removeEventListener` that you passed to `addEventListener`, otherwise the event listener will not be removed.

Note that it can't be a new function with the same code, it has to be actually the same object in memory, meaning you need to save the function to a variable and use that same variable in both add and remove.